src/storage/v3/shard_rsm.cpp

```diff
@@ -535,13 +535,17 @@ msgs::ReadResponses ShardRsm::HandleRead(msgs::GetPropertiesRequest &&req) {
     return result;
   };
 
-  auto collect_props = [&req](const VertexAccessor &v_acc,
-                              const std::optional<EdgeAccessor> &e_acc) -> ShardResult<std::map<PropertyId, Value>> {
+  auto collect_props = [this, &req](
+                           const VertexAccessor &v_acc,
+                           const std::optional<EdgeAccessor> &e_acc) -> ShardResult<std::map<PropertyId, Value>> {
     if (!req.property_ids) {
+      const auto *schema = shard_->GetSchema(shard_->PrimaryLabel());
+      MG_ASSERT(schema);
+
```

Comment:
First of all, I had to do the similar changes on the `ExpandOne` PR. However I would put this after the condition as for edges we don't need the schema. Pushing the instantiation of a local variable as late as ~possible~ reasonable is always a good thing.